subprojects/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionFactory.java

```diff
@@ -110,6 +117,69 @@ private static void showUsage(PrintStream out, CommandLineParser parser) {
         out.println();
     }
 
+    private static class WelcomeMessageAction implements Action<PrintStream> {
+        private final BuildLayoutParameters buildLayoutParameters;
+
+        public WelcomeMessageAction(BuildLayoutParameters buildLayoutParameters) {
+            this.buildLayoutParameters = buildLayoutParameters;
+        }
+
+        @Override
+        public void execute(PrintStream out) {
+            GradleVersion currentVersion = GradleVersion.current();
+            File gradleUserHomeDir = buildLayoutParameters.getGradleUserHomeDir();
+            File notificationsDir = new File(gradleUserHomeDir, "notifications/" + currentVersion.getVersion());
+            File markerFile = new File(notificationsDir, "release-features.rendered");
+
+            if (!markerFile.exists()) {
+                out.println();
+                out.print("Welcome to Gradle " + currentVersion.getVersion() + "!");
+
+                String featureList = readReleaseFeatures();
+
+                if (featureList != null) {
+                    out.println();
+                    out.println();
+                    out.print("Here is what's new:");
```

Comment:
I think this could easily be misinterpreted as "here's _all_ that's new". I would recommend we say something like:

> Here are the highlights of this release:

because it makes it clear that there is more, and that it's just for this release, not what is new since Gradle 4.1, say

Reply:
done